installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -22,68 +23,99 @@ enum Shell
+            try
             {
-                stateToolPath = session.CustomActionData["STATE_TOOL_PATH"];
-                session.Log("State Tool is installed, no installation required");
-                Status.ProgressBar.Increment(session, 1);
-                return ActionResult.Success;
+                WebClient client = new WebClient();
+                versionInfoString = client.DownloadString(jsonURL);
+            }
+            catch (WebException e)
+            {
+                string msg = string.Format("Encoutered exception downloading state tool json info file: {0}", e.ToString());
+                session.Log(msg);
+                ActiveState.RollbarHelper.Report(msg);
+                return ActionResult.Failure;
             }
 
-            string tempDir = Path.GetTempPath();
-            string scriptPath = Path.Combine(tempDir, "install.ps1");
-            string installPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "ActiveState", "bin");
-
-            Status.ProgressBar.StatusMessage(session, "Installing State Tool...");
+            VersionInfo info = JsonConvert.DeserializeObject<VersionInfo>(versionInfoString);
 
-            ServicePointManager.SecurityProtocol |= SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;
+            string windowsZip = "windows-amd64.zip";
+            string zipURL = stateURL + info.version + "/" + windowsZip;
+            string zipPath = Path.Combine(tempDir, windowsZip);
+            session.Log(string.Format("Downloading zip file from URL: {0}", zipURL));
+            Status.ProgressBar.StatusMessage(session, "Downloading State Tool...");
             try
             {
                 WebClient client = new WebClient();
-                client.DownloadFile("https://platform.activestate.com/dl/cli/install.ps1", scriptPath);
+                client.DownloadFile(zipURL, zipPath);
             }
             catch (WebException e)
             {
-                session.Log(string.Format("Encoutered exception downloading file: {0}", e.ToString()));
-                ActiveState.RollbarHelper.Report(string.Format("Encoutered exception downloading file: {0}", e.ToString()));
+                string msg = string.Format("Encoutered exception downloading state tool zip file: {0}", e.ToString());
+                session.Log(msg);
+                ActiveState.RollbarHelper.Report(msg);
                 return ActionResult.Failure;
             }
 
-            string installCmd = string.Format("Set-PSDebug -trace 2; Set-ExecutionPolicy Unrestricted -Scope Process; & '{0}' -n -t \"\"\"{1}\"\"\"", scriptPath, installPath);
-            session.Log(string.Format("Running install command: {0}", installCmd));
+            SHA256 sha = SHA256.Create();
+            FileStream fInfo = File.OpenRead(zipPath);
+            string zipHash = BitConverter.ToString(sha.ComputeHash(fInfo)).Replace("-", string.Empty).ToLower();
```

Comment:
What's the Replace for?

Reply:
The bit converter returns a hash with dashes (ie. `0F-AC-08...`) so we have to remove them to compare with the hash in the JSON file.